Keypad/Keypad.js

```diff
@@ -0,0 +1,327 @@
+ * @ui
+ * @public
+ */
+const Key = kind({
+	name: 'Key',
+
+	propTypes: /** @lends agate/Keypad.Key.prototype */ {
+		/**
+		 * Called when this button is clicked. Includes the 'key' key in its event payload to let the clicker know what was clicked inside their callback.
+		 *
+		 * @type {Function}
+		 * @param {Object} event
+		 * @public
+		 */
+		onKeyButtonClick: PropTypes.func,
+
+		/**
+		 * Text displayed below the number/icon/symbol on the key.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		subtext: PropTypes.string,
+
+		/**
+		 * Text displayed in the center of the key.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		text: PropTypes.string
+	},
+
+	styles: {
+		css,
+		className: 'key'
+	},
+
+	computed: {
+		textComponent: ({text}) => {
+			return (text != null && text !== '') ? <span className={css.text}>{text}</span> : null;
+		},
+		subtextComponent: ({subtext}) => {
+			return (subtext != null && subtext !== '') ? <span className={css.subtext}>{subtext}</span> : null;
+		}
+	},
+
+	handlers: {
+		onClick: handle(
+			forward('onClick'),
+			adaptEvent((ev, {children: key}) => ({key}), forward('onKeyButtonClick'))
+		)
+	},
+
+	render: ({children, subtextComponent, textComponent, ...rest}) => {
+		delete rest.onKeyButtonClick;
+
+		return (
+			<div className={css.keyContainer}>
+				<Button
+					{...rest}
+					css={css}
+					icon={children}
+					size="large"
+				>
+					{textComponent}
+					{subtextComponent}
+				</Button>
+			</div>
+		);
+	}
+});
+
+/**
+ * The basic Layout of a Keypad component.
+ *
+ * @class KeypadBase
+ * @memberof agate/Keypad
+ * @extends agate/Keypad.Key
+ * @ui
+ * @public
+ */
+const KeypadBase = kind({
+	name: 'Key',
+
+	propTypes: /** @lends agate/Keypad.KeypadBase.prototype */{
+		/**
+		 * Applies a disabled style and the control becomes non-interactive.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		disabled: PropTypes.bool,
+
+		/**
+		 * Called when a button is clicked. Includes the 'key' key in its event payload, updates the state and the input value accordingly.
+		 *
+		 * @type {Function}
+		 * @param {Object} event
+		 * @public
+		 */
+		handleInputValue: PropTypes.func
+	},
+
+	styles: {
+		css,
+		className: 'keypad'
+	},
+
+	render: ({disabled, handleInputValue, ...rest}) => {
+		return (
+			<Layout {...rest} align="center end" className={css.keypad} inline wrap>
+				{KEY_LIST.map((keyText, rowIndex) => {
+					return (
+						<Cell
+							aria-label={keyText.text === 'arrowleftturn' ? $L('Back Space') : keyText.text}
+							component={Key}
+							disabled={disabled}
+							key={`key${rowIndex}-${keyText.text}`}
+							onKeyButtonClick={() => handleInputValue(keyText.text)}
+							shrink
+							subtext={keyText.subtext}
+							text={keyText.text === 'arrowleftturn' || keyText.text === 'phone' ? null : keyText.text}
+						>
+							{keyText.text === 'arrowleftturn' || keyText.text === 'phone' ? keyText.text : null}
```

Comment:
This set of conditionals may become simpler if you incorporate an `icon` key into your `KEY_LIST` array object. Then you just pass the values in more directly: text->children, icon->icon, subtext->subtext.

I'd also recommend changing the term "subtext" to "label" which is less position-specific and allows for greater customizability. _(It's also a few letters shorter. hehe)_

Ex:
```javascript
const KEY_LIST = [
	{text: '1'},
	{text: '2', label: 'abc'},
	{text: '3', label: 'def'},
	{text: '4', label: 'ghi'},
	{text: '5', label: 'jkl'},
	{text: '6', label: 'mno'},
	{text: '7', label: 'pqrs'},
	{text: '8', label: 'tuv'},
	{text: '9', label: 'wxyz'},
	{text: '*'},
	{text: '0'},
	{text: '#'},
	{icon: 'phone',
	{icon: 'arrowleftturn'}
];
```